pkg/client/client.go

```diff
@@ -145,6 +145,16 @@ func (c ApplyClient) applyObject(ctx context.Context, obj client.Object, options
 	originalGeneration := existing.GetGeneration()
 	obj.SetResourceVersion(existing.GetResourceVersion())
 
+	// Special handling of ServiceAccounts is required because if a ServiceAccount is reapplied when it already exists, it causes Kubernetes controllers to
+	// automatically create new Secrets for the ServiceAccounts. After enough time the number of Secrets created will hit the Secrets quota and then no new
+	// Secrets can be created. To prevent this from happening, we keep the existing refs to secrets.
+	if strings.EqualFold(obj.GetObjectKind().GroupVersionKind().Kind, "ServiceAccount") {
+		MergeAnnotations(obj, existing.GetAnnotations()) // copy existing annotations
```

Comment:
shouldn't we merge also labels?
```suggestion
		MergeAnnotations(obj, existing.GetAnnotations()) // copy existing annotations
		MergeLabels(obj, existing.GetLabels())
```

Reply:
Makes sense, somehow I lost that. Done in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/cf4f268fefbf595507d343c3d97b36e593143235

Thanks!